### PR TITLE
Implement structured logging

### PR DESCRIPTION
**Pull request by togashidm**
Code updated with klogs and introduce the corresponding flag for the
klog level in TAS deployment.

### CODE REVIEW

File: extender/scheduler.go [hunk 1]
```diff
@@ -73,7 +74,7 @@ func handlerWithMiddleware(handle http.HandlerFunc) http.HandlerFunc {
 
 //error handler deals with requests sent to an invalid endpoint and returns a 404.
 func errorHandler(w http.ResponseWriter, r *http.Request) {
-	log.Print("unknown path")
+	klog.V(2).InfoS("Requested resource: '"+r.URL.Path+"' not found", "component", "extender")
```
Comment (killianmuldoon) [inline extender/scheduler.go:77]: Have you tested this variable use?

File: telemetry-aware-scheduling/pkg/cache/autoupdating.go [hunk 2]
```diff
@@ -81,7 +82,7 @@ func (n *AutoUpdatingCache) ReadMetric(metricName string) (metrics.NodeMetricsIn
 			return metric, nil
 		}
 	}
-	return metrics.NodeMetricsInfo{}, errors.New("no metric " + metricName + " found")
+	return metrics.NodeMetricsInfo{}, errors.New("no metric " + metricName + " found yet")
```
Comment (killianmuldoon) [inline telemetry-aware-scheduling/pkg/cache/autoupdating.go:85]: Remove this 'yet' the context isn't fully clear

File: telemetry-aware-scheduling/pkg/telemetryscheduler/telemetryscheduler.go [hunk 3]
```diff
@@ -159,15 +161,15 @@ func (m MetricsExtender) WritePrioritizeResponse(w http.ResponseWriter, result *
...
 	extenderArgs, err := m.DecodeExtenderRequest(r)
 	if err != nil {
-		log.Printf("cannot filter %v", err)
+		klog.V(2).InfoS("cannot filter "+err.Error(), "component", "extender")
 		return
 	}
 	filteredNodes := m.filterNodes(extenderArgs)
 	if filteredNodes == nil {
-		log.Print("No filtered nodes returned")
+		klog.V(2).InfoS("No filtered nodes returned", "http.status", http.StatusNotFound, "component", "extender")
```
Comment (killianmuldoon) [inline telemetry-aware-scheduling/pkg/telemetryscheduler/telemetryscheduler.go:172]: remove the status from the log here - it's not adding anything and will just confuse the reader. http status should only be for the http client

File: telemetry-aware-scheduling/pkg/cache/autoupdating.go (hunk 2 above)
Comment (togashidm) [reply telemetry-aware-scheduling/pkg/cache/autoupdating.go]: change made.

File: telemetry-aware-scheduling/pkg/telemetryscheduler/telemetryscheduler.go (hunk 3 above)
Comment (togashidm) [reply telemetry-aware-scheduling/pkg/telemetryscheduler/telemetryscheduler.go]: status removed.

File: extender/scheduler.go [hunk 4]
```diff
@@ -86,22 +87,23 @@ func (m Server) StartServer(port string, certFile string, keyFile string, caFile
...
 	if unsafe {
-		log.Printf("Extender Listening on HTTP  %v", port)
+		klog.V(2).InfoS("Extender Listening on HTTP "+port, "component", "extender")
 		err = http.ListenAndServe(":"+port, mx)
 	} else {
-		log.Printf("Extender Now Listening on HTTPS  %v", port)
 		srv := configureSecureServer(port, caFile)
 		srv.Handler = mx
-		log.Fatal(srv.ListenAndServeTLS(certFile, keyFile))
+		klog.V(2).InfoS("Extender Listening on HTTPS "+port, "component", "extender")
```
Comment (killianmuldoon) [inline extender/scheduler.go:95]: Added this line here as no print statement for listening was happening otherwise.

**Comment by killianmuldoon** [comment]
Great changes overall and I think we can merge this as it's at par in content with the older logging system, but with the addition of structured and levelled logs. Good work! There's still a lot more to do to potentially improve logging I think though - like printing out more contextual information on the pod being scheduled etc. But that's over and above what we had previously - changing out the native log for klog is a great first step.

/LGTM

**Pull request merged**